wgpu-hal/src/gles/queue.rs

```diff
@@ -936,7 +939,7 @@ impl crate::Queue<super::Api> for super::Queue {
         self.reset_state();
         for cmd_buf in command_buffers.iter() {
             if let Some(ref label) = cmd_buf.label {
-                self.shared.context.push_debug_group(
+                self.shared.context.lock().push_debug_group(
```

Comment:
We should probably be favoring locking once at the top of the function and then releasing at the bottom, I can't imagine context switches are terribly cheap.

Reply:
Yeah, I tried that, but the issue was that `self.process` takes `&mut self` and borrowing the `gl` lock at the top takes a reference to `&self.shared.context` so Rust didn't like having the mutable and immutable references to `self` at the same time. I'm not sure a good way around it other than making `process` not take self and moving everything that it nees into function arguments.

Is that what we want to do or is there a better way?

Reply:
since `shared` is an arc, we can do:
```rust
let shared = Arc::clone(&self.shared);
let context = shared.context.lock();
```